lib/pages/transactions_page/widgets/custom_sliver_delegate.dart

```diff
@@ -0,0 +1,169 @@
+                    opacity: shrinkPercentage,
+                    child: Padding(
+                      padding: const EdgeInsets.symmetric(horizontal: 8),
+                      child: _buildCollapsedWidget(context),
+                    ),
+                  ),
+              ],
+            ),
+          ),
+          const SizedBox(height: 8),
+          const Divider(
+            thickness: 1.0,
+            height: 1,
+            color: Colors.black,
+          ),
+        ],
+      ),
+    );
+  }
+
+  _buildExtendedWidget(BuildContext context) {
+    return ClipRect(
+      child: FittedBox(
+        fit: BoxFit.fitHeight,
+        alignment: Alignment.center,
+        child: SizedBox(
+          width: MediaQuery.of(context).size.width,
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              const SizedBox(height: 8),
+              TabBar(
+                controller: tabController,
+                labelColor: Colors.white,
+                indicatorSize: TabBarIndicatorSize.tab,
+                unselectedLabelColor: Color(0xFFB9BABC),
+                splashBorderRadius: BorderRadius.circular(50),
+                indicator: BoxDecoration(
+                  borderRadius: BorderRadius.circular(50),
+                  color: Color(0xFF356CA3),
+                ),
+                // TODO: capitalize text of the selected label
+                labelStyle: const TextStyle(fontWeight: FontWeight.bold),
+                unselectedLabelStyle:
+                    const TextStyle(fontWeight: FontWeight.normal),
+                tabs: myTabs,
+              ),
+              const SizedBox(height: 8),
+              // TODO: make sure buttons are not clickable when partially collapsed
+              MonthSelector(amount: amount),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  _buildCollapsedWidget(BuildContext context) {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+      children: [
+        Container(
+          padding: const EdgeInsets.symmetric(
+            vertical: 4.0,
+            horizontal: 8.0,
+          ),
+          decoration: BoxDecoration(
+            borderRadius: BorderRadius.circular(50),
+            color: Color(0xFF356CA3),
+          ),
+          child: Text(
+            myTabs[tabController.index].text!,
+            textAlign: TextAlign.center,
+            style: Theme.of(context)
+                .textTheme
+                .labelLarge
+                ?.copyWith(color: Colors.white),
+          ),
+        ),
+        Text(
+          "Settembre 2022",
+          style:
+              Theme.of(context).textTheme.displayMedium?.copyWith(fontSize: 14),
+        ),
+        Text(
+          "$amount €",
+          style: TextStyle(
+              color: (amount > 0) ? Colors.green : Colors.red,
+              fontWeight: FontWeight.bold,
+              fontFamily: 'SF Pro Text'),
```

Comment:
I'd suggest you to use a macro for the font name (and replace it where necessary, also in the other widgets)

I apologize if you do not know what a macro is, I mean something like a global variable.

Let's talk about whether it is better to create a new file for the macros.

(https://docs.flutter.dev/cookbook/design/fonts)

Reply:
I wasn't sure how we planned to handle fonts and colors across the app, so I temporarily hardcoded them.
One way would be to have a file where we define global variables (macros), but we could also define a Theme and make all widgets inherit from it.
I can use a macro but I don't know where to put it

Reply:
I think that at this project state, we could use a file and in the future make all widgets inherit from the theme. What do you think?

Reply:
Aren't fonts already defined here?
https://github.com/RIP-Comm/sossoldi/blob/20f13797f5bf5d578e66525892245020c2f00ccc/lib/main.dart#L35

Reply:
Done